grails-app/services/org/pih/warehouse/inventory/RefreshProductAvailabilityEventService.groovy

```diff
@@ -25,7 +25,12 @@ class RefreshProductAvailabilityEventService implements ApplicationListener<Refr
             return
         }
 
-        productAvailabilityService.triggerRefreshProductAvailability(event.locationId,
-                event.productIds, event.forceRefresh)
+        if (event?.synchronousRequired) {
+            productAvailabilityService.refreshProductsAvailability(event.locationId,
```

Comment:
Again be careful here. This action might be doing a lot more than the caller intended which might lead to performance implications.